### PR TITLE
refactor(core tokens): split core tokens into groups

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #8277

## Summary

This is a refactor only and will have no effect on existing output. Similar to Semantic tokens, this PR breaks the Core tokens into separate token type groups and puts them in a "core" token directory.

### CODE REVIEW

File: packages/calcite-design-tokens/src/index.ts [hunk 1]
```diff
@@ -17,19 +16,4 @@ export const globalTokens = await createCalciteTokenFiles({
   references: coreTokens.source,
 });
 
-export const componentTokens = await getFiles(resolve(__dirname, "component")).then((components) =>
```
Comment (jcfranco) [inline packages/calcite-design-tokens/src/index.ts:20]: Please correct me if I'm wrong, but this seems unrelated to the group changes. If so, we can either revert here or merge #8280 before merging this one.

**Review by jcfranco — COMMENTED** [review]
This should use the `refactor` commit type as it is not providing a fix.

File: packages/calcite-design-tokens/src/index.ts (hunk 1 above)
Comment (alisonailea) [reply packages/calcite-design-tokens/src/index.ts:20]: oop thanks for the catch

**Review by jcfranco — APPROVED** [review]
✨👥👥👥➡️👤👤👤✨
